storage/payload.py

```diff
@@ -62,20 +62,33 @@ def chunk_lambda() -> Any:
     def get_by_lines(self, range_start: int) -> Iterator[tuple[bytes, int, int, Optional[int]]]:
         original_range_start: int = range_start
 
+        is_gzipped: bool = False
+        is_b64encoded: bool = False
         try:
             base64_decoded = base64.b64decode(self._payload, validate=True)
             if not base64_decoded.startswith(b"\037\213"):  # gzip compression method
```

Comment:
nit: and not related to the PR - would be good to remove duplication of this check (L77, L111)

Reply:
I'm not sure we can remove duplicating the check
we have the following scenarios:
1- valid base64encode of gzip content
2- valid base64encode of plaintext content
3- bare gzip content
4- plaintext that will not raise a `binascii.Error` while base64 decoding but still does not produce valid unicode content.

4 it's tricky, and I've found totally by chance by sending some plaintext content (I don't remember exactly, whatever like "text blah blah"). this was base 64 encode to not unicode string, failing later on when try to decode in the decorators

that's why I added [L70](https://github.com/elastic/elastic-serverless-forwarder/pull/148/files/ad13910c8fc50a5865d8bd891806b48d6eae750b#diff-11a8f6bb928926e1b10ead6bb98a7adbea55f0590cb44c18d7094e7f0d883b15R70), but I have to skip if it's gzip content, because that will raise the `UnicodeDecodeError`

Reply:
was mainly just thinking it would be good to have `startswith(b"\037\213")` refactored into a function